### PR TITLE
docs: SDK architecture + small changes

### PR DESCRIPTION
**Pull request by szuperaz**
- SDK architecture page written
- @DaemonLoki I've changed some of your work around how to use the package, sorry about that, we are still thinking through some questions

### CODE REVIEW

**Review by oliverlaz — APPROVED** [review]

File: packages/react-sdk/docusaurus/docs/React/01-basics/02-introduction.mdx [hunk 1]
```diff
@@ -47,7 +38,8 @@ The fastest way to get started with the SDK is by trying the [Video Tutorial](./
 
 We know that everyone consumes documentation differently, which is why we want to make sure everyone can reach their goals as fast as possible.
 
-- Go through it sequentially and look at the next pages (heading to [Installation](./basics/03-installation) next) to get a thorough introduction that covers the most important topics and gets you up to speed in the quickest way.
+- If you haven't yet checked out our [Video Tutorial](./basics/01-tutorial), we suggest starting there.
```
Comment (MartinCupela) [inline packages/react-sdk/docusaurus/docs/React/01-basics/02-introduction.mdx:41]: @szuperaz this duplicates the note above

Comment (szuperaz) [reply packages/react-sdk/docusaurus/docs/React/01-basics/02-introduction.mdx:41]: I though it's worth mentioning it again as part of this "where to next" section

File: packages/react-sdk/docusaurus/docs/React/01-basics/02-introduction.mdx [hunk 2]
```diff
@@ -47,7 +38,8 @@ The fastest way to get started with the SDK is by trying the [Video Tutorial](./
 
 We know that everyone consumes documentation differently, which is why we want to make sure everyone can reach their goals as fast as possible.
```
Comment (MartinCupela) [inline packages/react-sdk/docusaurus/docs/React/01-basics/02-introduction.mdx:39]:
```suggestion
We know that everyone consumes documentation differently, which is why we want to make sure everyone can reach their goals as fast as possible. We suggest to you to:
```

File: packages/react-sdk/docusaurus/docs/React/01-basics/02-introduction.mdx (hunk 1 above)
Comment (MartinCupela) [reply packages/react-sdk/docusaurus/docs/React/01-basics/02-introduction.mdx:41]:
```suggestion
- checkout the introductory [Video Tutorial](./basics/01-tutorial) first
```

File: packages/react-sdk/docusaurus/docs/React/01-basics/02-introduction.mdx [hunk 3]
```diff
@@ -47,7 +38,8 @@ The fastest way to get started with the SDK is by trying the [Video Tutorial](./
...
 
-- Go through it sequentially and look at the next pages (heading to [Installation](./basics/03-installation) next) to get a thorough introduction that covers the most important topics and gets you up to speed in the quickest way.
+- If you haven't yet checked out our [Video Tutorial](./basics/01-tutorial), we suggest starting there.
+- Go through the introducation sequentially and look at the next pages (heading to [SDK architecture](./03-sdk-architecture.mdx) next) to get a thorough introduction that covers the most important topics and gets you up to speed in the quickest way.
 - Search for a specific problem using the search bar at the top, getting guided there directly.
```
Comment (MartinCupela) [inline packages/react-sdk/docusaurus/docs/React/01-basics/02-introduction.mdx:43]:
```suggestion
- search for a specific problem using the search bar at the top
```

File: packages/react-sdk/docusaurus/docs/React/01-basics/02-introduction.mdx [hunk 4]
```diff
@@ -47,7 +38,8 @@ The fastest way to get started with the SDK is by trying the [Video Tutorial](./
...
 - Explore our guides where we cover the more advanced topics that you might encounter, for example [integrating Video with Chat](./guides/06-chat-with-video), [Audio Rooms](./guides/08-audio-rooms), and [Livestreams](./guides/09-livestream).
```
Comment (MartinCupela) [inline packages/react-sdk/docusaurus/docs/React/01-basics/02-introduction.mdx:44]:
```suggestion
- explore our guides where we cover the more advanced topics, for example [integrating Video with Chat](./guides/06-chat-with-video), [Audio Rooms](./guides/08-audio-rooms), and [Livestreams](./guides/09-livestream).
```

File: packages/react-sdk/docusaurus/docs/React/01-basics/02-introduction.mdx (hunk 3 above)
Comment (MartinCupela) [reply packages/react-sdk/docusaurus/docs/React/01-basics/02-introduction.mdx:43]: Maybe we could add a element that when clicked focuses the search input?

File: packages/react-sdk/docusaurus/docs/React/01-basics/03-sdk-architecture.mdx [hunk 5]
```diff
@@ -3,4 +3,21 @@ id: 03-sdk-architecture
 title: SDK Architecture
 ---
 
-Stay tuned for more information on this topic.
+This page describes the main building blocks of the SDK.
+
+## Call Engine
+
+The call engine is made up of two parts:
```
Comment (MartinCupela) [inline packages/react-sdk/docusaurus/docs/React/01-basics/03-sdk-architecture.mdx:10]:
```suggestion
We understand the term **call engine** as a set of the following functionalities:
```

File: packages/react-sdk/docusaurus/docs/React/01-basics/03-sdk-architecture.mdx [hunk 6]
```diff
@@ -3,4 +3,21 @@ id: 03-sdk-architecture
...
+
+- The client lets you interact with Stream's video API
+- Various utility hooks and contexts that let you access the state easily and give high-level APIs for common video/audio call-related tasks such as device management
+
+Stream's video API supports two different call lifecycles. Our [Call lifecycle](../02-guides/02-call-lifecycle.gen.mdx) guide provides more information on the topic to help you decide which suits your requirements.
+
+Additionally, there are various available call types with configurations suiting different use cases (for example, livestream or audio room). If you want to know more about these, visit our [Configuring call defaults](./06-configuring-call-defaults.mdx) guide.
```
Comment (MartinCupela) [inline packages/react-sdk/docusaurus/docs/React/01-basics/03-sdk-architecture.mdx:17]: What does "call types with configurations" mean?

File: packages/react-sdk/docusaurus/docs/React/01-basics/03-sdk-architecture.mdx [hunk 7]
```diff
@@ -3,4 +3,21 @@ id: 03-sdk-architecture
...
+This page describes the main building blocks of the SDK.
+
+## Call Engine
+
+We understand the term **call engine** as a set of the following functionalities:
+
+- The client lets you interact with Stream's video API
+- Various utility hooks and contexts that let you access the state easily and give high-level APIs for common video/audio call-related tasks such as device management
+
+The SDK API supports two different call lifecycles. We have prepared a  [Call lifecycle guide](../02-guides/02-call-lifecycle.gen.mdx) to help you identify the right scenario for your project.
```
Comment (MartinCupela) [inline packages/react-sdk/docusaurus/docs/React/01-basics/03-sdk-architecture.mdx:15]:
```suggestion
The SDK API supports two different call lifecycles. We have prepared a [Call lifecycle guide](../02-guides/02-call-lifecycle.gen.mdx) to help you identify the right scenario for your project.
```

File: packages/react-sdk/docusaurus/docs/React/01-basics/03-sdk-architecture.mdx (hunk 6 above)
Comment (MartinCupela) [reply packages/react-sdk/docusaurus/docs/React/01-basics/03-sdk-architecture.mdx:17]:
```suggestion
Additionally, there are various available call types configurations suiting different use cases (for example, livestream or audio room). If you want to know more about these, visit our [Configuring call defaults](./06-configuring-call-defaults.mdx) guide.
```

File: packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx [hunk 8]
```diff
@@ -17,19 +17,19 @@ Here is a list of things still to address on this page:
 There are different ways to use the React Video SDK to build out UI. It comes with powerful built-in video components ready for use.
 In addition, it allows for styling and custom theming options with custom CSS.
 
-For custom use cases it is also possible to only use the low-level client to provide completely custom UI to fully tailor the SDK towards your needs.
+For custom use cases it is also possible to create your own components to fully tailor the SDK towards your needs. Please note that we suggest not to override a few, core components as those contain complex, low-level logic related to performance, the list of these components can be found in the [UI Components overview](../03-ui/01-overview.mdx).
 
 This page guides you through all of them with increasing levels of customization.
```
Comment (MartinCupela) [inline packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx:22]: What the adverb "them" refers to?

File: packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx [hunk 9]
```diff
@@ -17,19 +17,19 @@ Here is a list of things still to address on this page:
...
 
-### Using pre-built components
+## Using pre-built components
 
 The SDK comes with a lot of pre-built components for the different elements of the UI.
 
 :::tip
 For a list of all available UI components head over to the [Components](../../ui/01-ui-overview) section.
```
Comment (MartinCupela) [inline packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx:29]:
```suggestion
For a list of all available UI components head over to the [Components](../../03-ui/01-overview) section.
```

File: packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx (hunk 8 above)
Comment (szuperaz) [reply packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx:22]: hm, I can't seem to find "them" in the text above, if you mean the "the list of these" that refers to the core components referenced in the previous part of the sentence, it seemed to clear to me when I was writing it

Comment (MartinCupela) [reply packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx:22]: This page guides you through all of **them** with increasing levels of customization.

File: packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx [hunk 10]
```diff
@@ -14,22 +14,22 @@ Here is a list of things still to address on this page:
...
 :::
 
-### Theming with CSS
+## Theming with CSS
 
 If you only want to customize the styling of the available components, there is a wide range of CSS variables that you can override.
 With that, you can achieve a significant amount of theming in your video product.
```
Comment (MartinCupela) [inline packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx:35]: The following section has a grammatic mistake:

Here are a few examples: -> Here are few examples:

File: packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx (hunk 9 above)
Comment (szuperaz) [reply packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx:29]: The link works without the changes, I think there are two link formats: one is URL based and the other is file based

File: packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx (hunk 8 above)
Comment (szuperaz) [reply packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx:22]: Ah, I looked only to my changes, that line wasn't written by me, but from the context them refers to the different customization levels.

File: packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx [hunk 11]
```diff
@@ -14,22 +14,22 @@ Here is a list of things still to address on this page:
...
 
-For custom use cases it is also possible to only use the low-level client to provide completely custom UI to fully tailor the SDK towards your needs.
+For custom use cases it is also possible to create your own components to fully tailor the SDK towards your needs. Please note that we suggest not to override a few, core components as those contain complex, low-level logic related to performance, the list of these components can be found in the [UI Components overview](../03-ui/01-overview.mdx).
 
 This page guides you through all of them with increasing levels of customization.
 
-### Using pre-built components
+## Using pre-built components
 
 The SDK comes with a lot of pre-built components for the different elements of the UI.
```
Comment (MartinCupela) [inline packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx:26]: But their customizability will be limited as we do not plan to allow passing Components to props.

Comment (MartinCupela) [reply packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx:26]: I am not sure how they will be able to customize the pre-built components.

File: packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx (hunk 10 above)
Comment (szuperaz) [reply packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx:35]: I think this page, in general, will need some more work, as mentioned by @DaemonLoki. Feel free to add those changes @MartinCupela, or we can revisit them outside of this PR.

File: packages/react-sdk/docusaurus/docs/React/01-basics/02-introduction.mdx [hunk 12]
```diff
@@ -45,11 +36,12 @@ The fastest way to get started with the SDK is by trying the [Video Tutorial](./
 
 ## How to use our introduction
 
-We know that everyone consumes documentation differently, which is why we want to make sure everyone can reach their goals as fast as possible.
+We know that everyone consumes documentation differently, which is why we want to make sure everyone can reach their goals as fast as possible. We suggest to you to:
```
Comment (DaemonLoki) [inline packages/react-sdk/docusaurus/docs/React/01-basics/02-introduction.mdx:39]:
```suggestion
We know that everyone consumes documentation differently, so we want to ensure everyone can reach their goals as fast as possible. We suggest to you:
```

File: packages/react-sdk/docusaurus/docs/React/01-basics/02-introduction.mdx [hunk 13]
```diff
@@ -45,11 +36,12 @@ The fastest way to get started with the SDK is by trying the [Video Tutorial](./
...
 
-- Go through it sequentially and look at the next pages (heading to [Installation](./basics/03-installation) next) to get a thorough introduction that covers the most important topics and gets you up to speed in the quickest way.
-- Search for a specific problem using the search bar at the top, getting guided there directly.
-- Explore our guides where we cover the more advanced topics that you might encounter, for example [integrating Video with Chat](./guides/06-chat-with-video), [Audio Rooms](./guides/08-audio-rooms), and [Livestreams](./guides/09-livestream).
+- Checkout the introductory [Video Tutorial](./basics/01-tutorial) first.
+- Get acquainted with the [SDK architecture](./03-sdk-architecture.mdx) to get a thorough and quickiest introduction covering the most important topics.
+- Search for a specific problem using the search bar at the top.
+- Explore our guides where we cover the more advanced topics, for example [integrating Video with Chat](./guides/06-chat-with-video), [Audio Rooms](./guides/08-audio-rooms), and [Livestreams](./guides/09-livestream).
```
Comment (DaemonLoki) [inline packages/react-sdk/docusaurus/docs/React/01-basics/02-introduction.mdx:44]:
```suggestion
- Checkout the introductory [Video Tutorial](./basics/01-tutorial) first.
- Get acquainted with the [SDK architecture](./03-sdk-architecture.mdx) to get a thorough and quickest introduction covering the most important topics.
- Search for a specific problem using the search bar at the top.
- Explore our guides where we cover the more advanced topics, for example [integrating Video with Chat](./guides/06-chat-with-video), [Audio Rooms](./guides/08-audio-rooms), and [Livestreams](./guides/09-livestream).
```

File: packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx (hunk 8 above)
Comment (DaemonLoki) [reply packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx:22]: With more info in the paragraph before we should maybe change this for more clarity:

```suggestion
This page guides you through all the options you have with increasing levels of customization.
```

File: packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx [hunk 14]
```diff
@@ -14,22 +14,22 @@ Here is a list of things still to address on this page:
 
 ## Introduction
 
-There are different ways to use the React Video SDK to build out UI. It comes with powerful built-in video components ready for use.
+There are different ways to use the React Video SDK to build out UI. It comes with powerful built-in components ready for use.
 In addition, it allows for styling and custom theming options with custom CSS.
 
-For custom use cases it is also possible to only use the low-level client to provide completely custom UI to fully tailor the SDK towards your needs.
+For custom use cases it is also possible to create your own components to fully tailor the SDK towards your needs. Please note that we suggest not to override a few, core components as those contain complex, low-level logic related to performance, the list of these components can be found in the [UI Components overview](../03-ui/01-overview.mdx).
```
Comment (DaemonLoki) [inline packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx:20]:
```suggestion
For custom use cases it is also possible to create your own components to fully tailor the SDK to your needs. Please note that we suggest not to override a few, core components as those contain complex, low-level logic related to performance, the list of these components can be found in the [UI Components overview](../03-ui/01-overview.mdx).
```

File: packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx (hunk 11 above)
Comment (DaemonLoki) [reply packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx:26]: Some of them offer some props to tweak them. In any case there are options for styling them via CSS properties, right?

File: packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx (hunk 10 above)
Comment (DaemonLoki) [reply packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx:35]: Agree with Zita. 

It surely requires info (e.g. I didn't know about the `str-video` thing up until yesterday. So there's surely more things needed, I just wanted to give a basic skeleton to have a start.

Please always feel free to fill this with relevant wisdom! :)

Comment (DaemonLoki) [reply packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx:35]: > The following section has a grammatic mistake:
> 
> Here are a few examples: -> Here are few examples:

I'd argue that the first one is actually correct?

File: packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx (hunk 11 above)
Comment (MartinCupela) [reply packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx:26]: I would conside the CSS part on the following section. I am still not sure how customizable they will be. Our intention is not to follow the stream-chat-react pattern, where you pass down components in props.

File: packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx (hunk 10 above)
Comment (MartinCupela) [reply packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx:35]: few vs. a few (https://www.grammarly.com/blog/few-a-few/) - not sure which you had in mind @DaemonLoki

File: packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx (hunk 11 above)
Comment (MartinCupela) [reply packages/react-sdk/docusaurus/docs/React/01-basics/07-ui-customization-options.mdx:26]: Ok, let's revisit this later

File: packages/react-sdk/docusaurus/docs/React/01-basics/02-introduction.mdx (hunk 3 above)
Comment (DaemonLoki) [reply packages/react-sdk/docusaurus/docs/React/01-basics/02-introduction.mdx]: I'd keep capitalization the same across all bullet points. So either have all as sentences starting with capital letters and punctuation or change all of them.
The wording is something where you know better how to precisely formulate things.

**Pull request merged**